crml/transaction-payment/src/lib.rs

```diff
@@ -50,7 +51,7 @@ type Multiplier = Fixed64;
 type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
 type NegativeImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::NegativeImbalance;
 
-pub trait Trait: system::Trait {
+pub trait Trait: system::Trait + generic_asset::Trait {
```

Comment:
Can add `type Balance: ... ` and `type AssetId: ...` to the `transaction_payment::Trait` instead of the full dependency on `generic_assset`